src/services/InputCapture.py

```diff
@@ -0,0 +1,16 @@
+from src.shared_memory import SharedMemory
+from src.pin.pinout import Pinout
+import src.pin.memory as memory
+from ctypes import c_uint32
+from enum import Enum, auto, unique
+class InputCapture:
+    def __init__(self,pin1: Pinout):
+            #make the connection between the shared_memory and the pin
+            self._pin1 = SharedMemory.get_pin(pin1,memory.InputCapture)
+
+    def set_duty_cycle(self, duty_cycle: int) -> None:
+        self._pin1.data.duty_cycle = duty_cycle
+        
+    
+    def set_frequency(self,frequency: int) -> None:
+        self._pin1.data.frequency = frequency
```

Comment:
this is a PWM though? the microcontroller sees a PWM 
input capture should read PWMs, this will be useful when we add more functionality into the automated tests.

Reply:
Yes but as I understood, the input_capture from c++ reads PWM so from python we write in the shared_memory to change the values of the pwm that is reading the c++ input_capture

Reply:
yes